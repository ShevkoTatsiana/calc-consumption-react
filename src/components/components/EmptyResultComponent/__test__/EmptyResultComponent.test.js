import React from 'react';
import ReactDom from 'react-dom';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import {EmptyResultComponent} from '../EmptyResult.component';

afterEach(cleanup);
it('renders without crashing', ()=> {
    const div = document.createElement('div');
    ReactDom.render(<EmptyResultComponent></EmptyResultComponent>, div);
});

it('renders component correctly', ()=> {
    const {getByTestId} = render(<EmptyResultComponent/>);
    expect(getByTestId('empty-result')).toHaveTextContent('Sorry, you don\'t have any active result');
});

it('match snapshot', ()=> {
    const tree = renderer.create(<EmptyResultComponent/>).toJSON();
    expect(tree).toMatchSnapshot();
});