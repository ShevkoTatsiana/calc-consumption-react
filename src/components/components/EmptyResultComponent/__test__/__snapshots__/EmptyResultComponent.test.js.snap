// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  data-testid="empty-result"
>
  Sorry, you don't have any active result
</div>
`;
